### PR TITLE
journald: use memdup_suffix0() when copying string from potentially b…

### PR DESCRIPTION
**Pull request by poettering**
…inary data

Fixes: #12484

### CODE REVIEW

**Comment by evverx** [comment]
@mrc0mmand I'm wondering if you can remember how you triggered that issue so that the test suite on CentOS CI/Travis CI can be updated accordingly?

**Comment by mrc0mmand** [comment]
@evverx we dropped the `-Dslow-tests=true` option from both [clang](https://github.com/systemd/systemd-centos-ci/blob/master/vagrant/Vagrantfiles/Vagrantfile_arch-sanitizers-clang) and [gcc](https://github.com/systemd/systemd-centos-ci/blob/master/vagrant/Vagrantfiles/Vagrantfile_arch-sanitizers-gcc) Vagrantfiles, so we should put it back once everything is fixed

File: src/journal/journald-native.c [hunk 1]
```diff
@@ -63,7 +63,7 @@ static void server_process_entry_meta(
                  startswith(p, "SYSLOG_IDENTIFIER=")) {
                 char *t;
 
-                t = strndup(p + 18, l - 18);
+                t = memdup_suffix0(p + 18, l - 18);
```
Comment (evverx) [inline src/journal/journald-native.c:66]: I should probably take a look at how t is used. If it's considered a string everywhere else we still lose the bytes after the first null byte in the middle of it. Maybe it should be escaped.

Comment (evverx) [reply src/journal/journald-native.c]: Oh, sorry. It just copies "SYSLOG_IDENTIFIER=" :-)

Comment (evverx) [reply src/journal/journald-native.c:66]: OK. I managed to send "MESSAGE":[104,101,121,0,104,111]," and "SYSLOG_IDENTIFIER":[72,69,89,0,72,79]" with `printf 'MESSAGE=hey\0ho\nSYSLOG_IDENTIFIER=HEY\0HO\n' | nc -w1 -u -U /run/systemd/journal/socket` so it seems to be working.

**Pull request merged**